compiler/util/files.cpp

```diff
@@ -891,6 +891,14 @@ void codegen_makefile(fileinfo* mainfile, const char** tmpbinname,
 
   // List object files needed to compile this deliverable.
   fprintf(makefile.fptr, "CHPLUSEROBJ = \\\n");
+  if (!fLibraryCompile) {
+    const char* dot = &(mainfile->pathname[strlen(mainfile->pathname)-2]);
+    const char* ext = &(mainfile->pathname[strlen(mainfile->pathname)-1]);
+    if (*dot != '.') {
+      INT_FATAL("Unexpected extension in mainfile for non-library compile");
+    }
```

Comment:
I read through your comments but can you explain why this check was needed?

Reply:
It seems that the CHPLSRC Makefile variable lists a `.c` file for a C back-end compilation, and a `.o` file for LLVM compilations, so this was a way to update the Makefile rule to work in either case.  I'm currently seeing if there's something more fundamental we can do w.r.t. these variables to make this feel less fragile.

Reply:
Oh, I see. But this actually checks whether we have a 1-char extension? Maybe that's the fragility that you are talking about. Adding `|| *ext != 'o' || *ext != 'c'` to the check would go a long way in improving the code. At least in terms of being more self-documenting, if not for fragility.

Reply:
Good point.  The main fragility I feel unhappy about is the need to handle both these cases simultaneously to begin with rather than having the Makefiles use different variables for different roles, but that's obviously a pre-existing condition.  The workaround for the library modes felt similarly like dancing on eggshells, although it's arguably for pre-existing reasons as well.

Reply:
Made your change in the latest commit.